pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -453,7 +455,10 @@ func (a AKS) applyOptions(nbv *NodeBootstrapVariables) {
 	nbv.ResourceGroup = a.ResourceGroup
 	nbv.UserAssignedIdentityID = a.KubeletIdentityClientID
 
-	nbv.NetworkPlugin = a.NetworkPlugin
+	// if we are using overlay, we delegate the installation of the cni to the cns daemonset and will put none as to not drop the
+	// 10-azure.conflist that will use nodesubnet
+	nbv.NetworkPlugin = lo.Ternary(a.NetworkPluginMode == consts.NetworkPluginModeOverlay, consts.NetworkPluginNone, a.NetworkPlugin)
+
```

Comment:
Follow-up here is if we are using 1.32 we will also set network plugin to None, for now just fixing for all existing karpenter clusters.